substrate/pkg/controller/substrate/cluster/addons/prometheusstack.go

```diff
@@ -0,0 +1,97 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+
+	"github.com/awslabs/kubernetes-iteration-toolkit/substrate/pkg/apis/v1alpha1"
+	"github.com/awslabs/kubernetes-iteration-toolkit/substrate/pkg/utils/helm"
+	"github.com/awslabs/kubernetes-iteration-toolkit/substrate/pkg/utils/kubectl"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type PrometheusStack struct{}
```

Comment:
```suggestion
type Prometheus struct{}
```

Reply:
I went back and forth between the two. Since, we are deploying not just Prometheus but also p-operator, grafana. So went with `PrometheusStack`